javascript/javascript2/week2/main.js

```diff
@@ -0,0 +1,129 @@
+// Doubling of number
+
+const numbers = [1, 2, 3, 4];
+const newNumbers = numbers
+  .filter((x) => x % 2 !== 0)
+  .map((filteredNumber) => filteredNumber * 2);
+console.log(newNumbers);
+
+// Create an array  containing the movies with a short title
+
+const shortMoviesTitle = movies.filter((movie) => !movie.title.includes(" "));
+console.log(shortMoviesTitle);
+
+// Create an array of movie titles with long movie titles
+
+const longMoviesTitles = movies
+  .filter((movie) => movie.title.includes(" "))
+  .map((filteredMovie) => filteredMovie.title);
+console.log(longMoviesTitles);
+
+// Count the number of movies made between 1980-1989 (including both the years)
+
+const numberOfMoviesMadeBetweenYears = movies.reduce((movieCounter, movie) => {
+  if (movie.year >= 1980 && movie.year <= 1989) {
+    return (movieCounter += 1);
+  }
+  return movieCounter;
+}, 0);
+
+console.log(numberOfMoviesMadeBetweenYears);
+
+// Create a new array that has an extra key called tag. The tag is based on the rating: Good (>= 7), Average (>= 4 and < 7), Bad (< 4)
+
+const tagedMovies = movies.map((movie) => {
+  if (movie.rating >= 7) {
+    movie.tag = "Good";
+    return movie;
+  } else if (movie.rating >= 4) {
+    movie.tag = "Average";
+    return movie;
+  } else {
+    movie.tag = "Bad";
+    return movie;
+  }
+});
+console.log(tagedMovies);
+
+// Using chaining, first filter the movies array to only contain the movies rated higher than 6.
+// Now map the movies array to only the rating of the movies.
+
+const ratedMovies = movies
+  .filter((movie) => movie.rating > 6)
+  .map((filteredMovie) => {
+    return filteredMovie.rating;
+  });
```

Comment:
Excellent! 
 
One comment, you can do this `map` statement as a one-liner:  `.map(filteredMovie => filteredMovie.rating)`